lighthouse-cli/osx-chrome.js

```diff
@@ -0,0 +1,114 @@
+'use strict';
+
+const spawn = require('child_process').spawn;
+const through2 = require('through2');
+const path = require('path');
+const fs = require('fs');
+
+function canAccess(file) {
+  try {
+    fs.accessSync(file);
+    return true;
+  } catch (e) {
+    return false;
+  }
+}
+
+class LsRegisterError extends Error {
+  constructor(code, desc) {
+    super(`LsRegister exited with ${code}`);
+    this.code = code;
+    this.desc = desc;
+  }
+}
+
+class OsxChrome {
+  constructor() {
+    this.LSREGISTER =
+      '/System/Library/Frameworks/CoreServices.framework' +
+      '/Versions/A/Frameworks/LaunchServices.framework' +
+      '/Versions/A/Support/lsregister';
+    this.CHROME_REGEX = /^path: (.*)google chrome canary.app$/;
+    this.isLsRegisterError = err => err instanceof LsRegisterError;
+    this.suffix = '/Contents/MacOS/Google Chrome Canary';
+  }
+  sortChromeInstallations(installations) {
+    const priorities = new Map([
+      [/^\/Volumes\//, -1],
+      [/^\/Applications\//, 100],
+      [new RegExp(`^${process.env.HOME}/Applications/`), 50]
+    ]);
+    const defaultPriority = 10;
+
+    return installations
+      // assign priorities
+      .map(inst => {
+        for (let [regex, priority] of priorities) {
+          if (regex.test(inst)) {
+            return [inst, priority];
+          }
+        }
+        return [inst, defaultPriority];
+      })
+      // sort based on priorities
+      .sort((a, b) => {
+        if (a[1] < b[1]) {
+          return 1;
+        }
+        if (a[1] > b[1]) {
+          return -1;
+        }
+        return 0;
+      })
+      // remove priority flag
+      .map(pair => pair[0]);
+  }
+
+  getChrome() {
+    return new Promise((resolve, reject) => {
+      const dump = spawn(this.LSREGISTER, ['-dump']);
```

Comment:
you'd save a good number of lines of code (and be able to get rid of the `through2` dep) by switching to something like
```javascript
execSync(`${this.LSREGISTER} -dump | grep -i "google chrome canary.app$"`).toString()
```